packages/playground/src/util/cacheHandlers.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  cloneElementInCode,
+  deleteElementInCode,
+  findElementInAST,
+} from './codeHandlers';
+
+const getDataFromCache = async (
+  file: string,
+  char: number,
+): Promise<{ body?: string; cache?: Cache; charCount?: number }> => {
+  try {
+    const cache = await caches.open('playground');
```

Comment:
Proper error handling https://medium.com/front-end-weekly/error-propagation-in-javascript-with-error-translation-pattern-78cf7178fe92